pc2cps

```diff
@@ -72,18 +86,60 @@ class pkgconfig(object):
             else:
                 deps[dep[0]] = {}
 
-        return deps
+        self.requires = deps
 
     #--------------------------------------------------------------------------
     @property
     def component(self):
         def _keys(x):
-            return None if x is None else x.keys()
-
-        c = {
-            'Type': 'interface',
-        }
-        extend(c, 'Link-Flags', self._get_split('Libs'))
+            return [] if x is None else x.keys()
+
+        def _list(re_):
+            ma = [re_.match(x) for x in link_flags]
+            return [x.group(1) for x in ma if x]
+
+        # If not found and if package name given in library list with
+        # "-l" flag, an error should be thrown?
+        def _find_location():
+            for path in self._paths:
+                for file in os.listdir(path):
+                    pattern = '(lib)?' + self.library_name + self._lib_pattern
+                    ma = re.match(pattern, file)
+                    if ma is not None:
+                        return os.path.join(path, ma.group(0))
+
+        # Preprocess libraries
+        link_flags = self._get_split('Libs')
+        # Find all library paths
+        self._paths = [x for x in _list(self._re_libpath) if os.path.isdir(x)]
+        # Find all libraries
+        self.libname = _list(self._re_libname)
+        c = {}
```

Comment:
Please give this a more descriptive name

Reply:
I renamed it to `component`.